platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -284,15 +286,37 @@ private void captureTimestamp(final ConsensusEvent event) {
      * @param transaction the transaction to apply
      */
     private void handleTransaction(final ConsensusTransaction transaction) {
-        if (transaction.isSystem()) {
-            return;
+        int delta;
+
+        if (!transaction.getTransaction().equals(EventTransaction.DEFAULT)) {
+            if (transaction.isSystem()) {
+                return;
+            }
+
+            delta = ByteUtils.byteArrayToInt(
+                    transaction.getApplicationTransaction().toByteArray(), 0);
+
+        } else {
+            if (isSystemTransaction(transaction.getTransactionsBytes())) {
+                return;
+            }
+
+            delta = ByteUtils.byteArrayToInt(transaction.getTransactionsBytes().toByteArray(), 0);
         }
-        final int delta =
-                ByteUtils.byteArrayToInt(transaction.getApplicationTransaction().toByteArray(), 0);
+
         runningSum += delta;
         setChild(RUNNING_SUM_INDEX, new StringLeaf(Long.toString(runningSum)));
     }
 
+    private boolean isSystemTransaction(final Bytes transactionBytes) {
+        if (transactionBytes == null || transactionBytes.length() == 0) {
+            return false;
+        }
+
+        final var transaction = transactionBytes.toByteArray();
+        return transaction[0] != 0;
+    }
+
```

Comment:
this doesnt work in isolation, transactions will have to be constructed differently (discussed)

Reply:
Updated